google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/strategy/CreateNodesWithGroupEncodedIntoNameThenAddToSet.java

```diff
@@ -131,12 +130,20 @@
    }
 
    /**
-    * Try and find a network either previously created by jclouds or user
-    * defined.
+    * Try and find a network previously created by the user.
     */
-   private Network getOrCreateNetwork(GoogleComputeEngineTemplateOptions templateOptions, String sharedResourceName) {
-      String networkName = templateOptions.network() != null ? toName(templateOptions.network()) : sharedResourceName;
-      return networkMap.getUnchecked(NetworkAndAddressRange.create(networkName, DEFAULT_INTERNAL_NETWORK_RANGE, null));
+   private Network getNetwork(Set<String> networks) {
+      String networkName;
+      if (networks == null || networks.isEmpty()){
+         networkName = DEFAULT_NETWORK_NAME;
+      }
+      else {
+         Iterator<String> iterator = networks.iterator();
+         networkName = nameFromNetworkString(iterator.next());
+         assert !iterator.hasNext() : "Error: Please specify only one network in TemplateOptions when using GCE.";
```

Comment:
Out of curiosity: why throwing an AssertionError instead of an IllegalArgumentException, or any other runtime exception?